### PR TITLE
Add slack notification for canary build failures

### LINKED ISSUE #42203: Inform when "canary" build fails
When our "canary" build fails in main, we only see it failed in actions: ( "scheduled" run currently) and some PRs

We should have a way to notify the committers that main is faling and needs fixing (Slack?/ Devlist?)

### PR DESCRIPTION
**Pull request by romsharon98**
closes: https://github.com/apache/airflow/issues/42203

this PR should send message to this slack channel https://apache-airflow.slack.com/archives/C07MWFASWJ1 if the CI is failed.

after verification on the main branch we need to change the alerts to be sent to the internal-ci-cd channel.

image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by jscheffl — APPROVED** [review]
Looks good - except small static checks. I assume we need to merge to test...
If not working we need a follow-up fix.
I propose to merge and see after fixing static checks.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -669,3 +670,32 @@ jobs:
...
+  notify-slack-failure:
+    name: "Notify Slack on Failure"
+    if: github.event_name == 'schedule' && failure()
+    runs-on: ["ubuntu-22.04"]
+    steps:
+      - name: Notify Slack
+        id: slack
+        uses: slackapi/slack-github-action@v1.27.0
+        with:
+          payload-delimiter: "."
```
Comment (jedcunningham) [inline .github/workflows/ci.yml]: Do we need this? This is the default, I also don't see where we are using it?

**Review by jedcunningham — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (romsharon98) [reply .github/workflows/ci.yml]: We don't. removed

**Review by shahar1 — APPROVED** [review]
Nice!

**Comment by jscheffl** [comment]
Seems we have a bug in the logic... with my limited Github actions knowledge... I assume dependecy is missing and therefore it is skipped as no failure is made at point of the task is activated and then decides to skip.

image

**Pull request merged**